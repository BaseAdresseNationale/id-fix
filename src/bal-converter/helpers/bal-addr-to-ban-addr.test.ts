import type { BalAdresse } from "../../types/bal-types.js";
import type { BanAddress } from "../../types/ban-types.js";

import { describe, expect, test } from "vitest";
import {
  numberForTopo as IS_TOPO_NB,
} from "../bal-converter.config.js";
import { idSampleWithAddressId, 
  idSampleWithMainTopoId,
  idSampleWithAddressIdAndMainTopoId,
  idSampleWithAllIds
} from "./__mocks__/fake-data.js";
import balAddrToBanAddr from "./bal-addr-to-ban-addr";

const defaultTestBalAddress: BalAdresse = {
  cle_interop: "21286_0001_00001",
  commune_insee: "21286",
  commune_nom: "Cocorico",
  voie_nom: "Route de la Baleine",
  numero: 1,
  position: "entrée",
  x: 1,
  y: 2,
  long: 1,
  lat: 2,
  date_der_maj: new Date ("2021-01-01"),
  certification_commune: true,
  source: "BAL",
};

describe("balAddrToBanAddr", () => {
  test("should return BanAddress without BanID & BanTopoID", async () => {
    expect(balAddrToBanAddr(defaultTestBalAddress)).toMatchSnapshot();
  });

  test("should return BanAddress with BanID without BanTopoID", async () => {
    const testBalAddress: BalAdresse = {
      ...defaultTestBalAddress,
      uid_adresse: idSampleWithAddressId,
    };

    expect(balAddrToBanAddr(testBalAddress)).toMatchSnapshot();
  });

  test("should return BanAddress with BanTopoID", async () => {
    const testBalAddress: BalAdresse = {
      ...defaultTestBalAddress,
      uid_adresse: idSampleWithMainTopoId,
    };

    expect(balAddrToBanAddr(testBalAddress)).toMatchSnapshot();
  });

  test("should return BanAddress with BanID and BanTopoID", async () => {
    const testBalAddress: BalAdresse = {
      ...defaultTestBalAddress,
      uid_adresse: idSampleWithAddressIdAndMainTopoId,
    };

    expect(balAddrToBanAddr(testBalAddress)).toMatchSnapshot();
  });

  test("should return BanAddress with BanID, BanTopoID, other toponyms and BanDistrictID", async () => {
    const testBalAddress: BalAdresse = {
      ...defaultTestBalAddress,
      uid_adresse: idSampleWithAllIds,
    };

    expect(balAddrToBanAddr(testBalAddress)).toMatchSnapshot();
  });

  test("should not consider as Ban Address", async () => {
    const testBalAddress: BalAdresse = {
      ...defaultTestBalAddress,
      uid_adresse: idSampleWithMainTopoId,
      numero: Number(IS_TOPO_NB),
    };

    expect(balAddrToBanAddr(testBalAddress)).toMatchSnapshot();
  });

  test("should return BanAddress with multiple positions", async () => {
    const testOldBalAddress: BalAdresse = {
      ...defaultTestBalAddress,
      uid_adresse: idSampleWithAllIds,
      position: "entrée",
      x: 1,
      y: 2,
      long: 1,
      lat: 2,
    };
    const oldBanAddress = balAddrToBanAddr(testOldBalAddress) as BanAddress;
    const testBalAddress: BalAdresse = {
      ...defaultTestBalAddress,
      uid_adresse: idSampleWithAllIds,
      position: "autre",
      x: 3,
      y: 4,
      long: 3,
      lat: 4,
    };

    expect(balAddrToBanAddr(testBalAddress, oldBanAddress)).toMatchSnapshot();
  });
});
