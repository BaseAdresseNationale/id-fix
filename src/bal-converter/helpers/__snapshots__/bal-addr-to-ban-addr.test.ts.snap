// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`balAddrToBanAddr > should not consider as Ban Address 1`] = `undefined`;

exports[`balAddrToBanAddr > should return BanAddress with BanID and BanTopoID 1`] = `
{
  "certified": true,
  "districtID": undefined,
  "id": "aaaaaaaa-0000-4aaa-9000-1234567890aa",
  "mainCommonToponymID": "bbbbbbbb-0000-4aaa-9000-1234567890bb",
  "meta": {
    "bal": {
      "cleInterop": "21286_0001_00001",
    },
  },
  "number": 1,
  "positions": [
    {
      "geometry": {
        "coordinates": [
          1,
          2,
        ],
        "type": "Point",
      },
      "type": "entrance",
    },
  ],
  "secondaryCommonToponymIDs": undefined,
  "updateDate": 2021-01-01T00:00:00.000Z,
}
`;

exports[`balAddrToBanAddr > should return BanAddress with BanID without BanTopoID 1`] = `
{
  "certified": true,
  "districtID": undefined,
  "id": "aaaaaaaa-0000-4aaa-9000-1234567890aa",
  "mainCommonToponymID": undefined,
  "meta": {
    "bal": {
      "cleInterop": "21286_0001_00001",
    },
  },
  "number": 1,
  "positions": [
    {
      "geometry": {
        "coordinates": [
          1,
          2,
        ],
        "type": "Point",
      },
      "type": "entrance",
    },
  ],
  "secondaryCommonToponymIDs": undefined,
  "updateDate": 2021-01-01T00:00:00.000Z,
}
`;

exports[`balAddrToBanAddr > should return BanAddress with BanID, BanTopoID, other toponyms and BanDistrictID 1`] = `
{
  "certified": true,
  "districtID": "dddddddd-0000-4aaa-9000-1234567890ff",
  "id": "aaaaaaaa-0000-4aaa-9000-1234567890aa",
  "mainCommonToponymID": "bbbbbbbb-0000-4aaa-9000-1234567890bb",
  "meta": {
    "bal": {
      "cleInterop": "21286_0001_00001",
    },
  },
  "number": 1,
  "positions": [
    {
      "geometry": {
        "coordinates": [
          1,
          2,
        ],
        "type": "Point",
      },
      "type": "entrance",
    },
  ],
  "secondaryCommonToponymIDs": [
    "cccccccc-0000-4aaa-9000-1234567890cc",
    "cccccccc-1111-4aaa-9000-1234567890dd",
    "cccccccc-2222-4aaa-9000-1234567890ee",
  ],
  "updateDate": 2021-01-01T00:00:00.000Z,
}
`;

exports[`balAddrToBanAddr > should return BanAddress with BanTopoID 1`] = `
{
  "certified": true,
  "districtID": undefined,
  "id": undefined,
  "mainCommonToponymID": "bbbbbbbb-0000-4aaa-9000-1234567890bb",
  "meta": {
    "bal": {
      "cleInterop": "21286_0001_00001",
    },
  },
  "number": 1,
  "positions": [
    {
      "geometry": {
        "coordinates": [
          1,
          2,
        ],
        "type": "Point",
      },
      "type": "entrance",
    },
  ],
  "secondaryCommonToponymIDs": undefined,
  "updateDate": 2021-01-01T00:00:00.000Z,
}
`;

exports[`balAddrToBanAddr > should return BanAddress with label 1`] = `
{
  "certified": true,
  "districtID": "dddddddd-0000-4aaa-9000-1234567890ff",
  "id": "aaaaaaaa-0000-4aaa-9000-1234567890aa",
  "labels": [
    {
      "isoCode": "fra",
      "value": "Ancien chemin du hero",
    },
  ],
  "mainCommonToponymID": "bbbbbbbb-0000-4aaa-9000-1234567890bb",
  "meta": {
    "bal": {
      "cleInterop": "21286_0001_00001",
    },
  },
  "number": 1,
  "positions": [
    {
      "geometry": {
        "coordinates": [
          1,
          2,
        ],
        "type": "Point",
      },
      "type": "entrance",
    },
  ],
  "secondaryCommonToponymIDs": [
    "cccccccc-0000-4aaa-9000-1234567890cc",
    "cccccccc-1111-4aaa-9000-1234567890dd",
    "cccccccc-2222-4aaa-9000-1234567890ee",
  ],
  "updateDate": 2021-01-01T00:00:00.000Z,
}
`;

exports[`balAddrToBanAddr > should return BanAddress with multilingual label 1`] = `
{
  "certified": true,
  "districtID": "dddddddd-0000-4aaa-9000-1234567890ff",
  "id": "aaaaaaaa-0000-4aaa-9000-1234567890aa",
  "labels": [
    {
      "isoCode": "fra",
      "value": "Ancien chemin du hero",
    },
    {
      "isoCode": "eus",
      "value": "Antzinako Heroiaren Bidea",
    },
  ],
  "mainCommonToponymID": "bbbbbbbb-0000-4aaa-9000-1234567890bb",
  "meta": {
    "bal": {
      "cleInterop": "21286_0001_00001",
    },
  },
  "number": 1,
  "positions": [
    {
      "geometry": {
        "coordinates": [
          1,
          2,
        ],
        "type": "Point",
      },
      "type": "entrance",
    },
  ],
  "secondaryCommonToponymIDs": [
    "cccccccc-0000-4aaa-9000-1234567890cc",
    "cccccccc-1111-4aaa-9000-1234567890dd",
    "cccccccc-2222-4aaa-9000-1234567890ee",
  ],
  "updateDate": 2021-01-01T00:00:00.000Z,
}
`;

exports[`balAddrToBanAddr > should return BanAddress with multilingual label after cleaning non-compliant ISO code lang 1`] = `
{
  "certified": true,
  "districtID": "dddddddd-0000-4aaa-9000-1234567890ff",
  "id": "aaaaaaaa-0000-4aaa-9000-1234567890aa",
  "labels": [
    {
      "isoCode": "fra",
      "value": "Ancien chemin du hero",
    },
  ],
  "mainCommonToponymID": "bbbbbbbb-0000-4aaa-9000-1234567890bb",
  "meta": {
    "bal": {
      "cleInterop": "21286_0001_00001",
    },
  },
  "number": 1,
  "positions": [
    {
      "geometry": {
        "coordinates": [
          1,
          2,
        ],
        "type": "Point",
      },
      "type": "entrance",
    },
  ],
  "secondaryCommonToponymIDs": [
    "cccccccc-0000-4aaa-9000-1234567890cc",
    "cccccccc-1111-4aaa-9000-1234567890dd",
    "cccccccc-2222-4aaa-9000-1234567890ee",
  ],
  "updateDate": 2021-01-01T00:00:00.000Z,
}
`;

exports[`balAddrToBanAddr > should return BanAddress with multilingual label and ignore empty default value 1`] = `
{
  "certified": true,
  "districtID": "dddddddd-0000-4aaa-9000-1234567890ff",
  "id": "aaaaaaaa-0000-4aaa-9000-1234567890aa",
  "labels": [
    {
      "isoCode": "fra",
      "value": "Ancien chemin du hero",
    },
  ],
  "mainCommonToponymID": "bbbbbbbb-0000-4aaa-9000-1234567890bb",
  "meta": {
    "bal": {
      "cleInterop": "21286_0001_00001",
    },
  },
  "number": 1,
  "positions": [
    {
      "geometry": {
        "coordinates": [
          1,
          2,
        ],
        "type": "Point",
      },
      "type": "entrance",
    },
  ],
  "secondaryCommonToponymIDs": [
    "cccccccc-0000-4aaa-9000-1234567890cc",
    "cccccccc-1111-4aaa-9000-1234567890dd",
    "cccccccc-2222-4aaa-9000-1234567890ee",
  ],
  "updateDate": 2021-01-01T00:00:00.000Z,
}
`;

exports[`balAddrToBanAddr > should return BanAddress with multilingual label and overwrited default value 1`] = `
{
  "certified": true,
  "districtID": "dddddddd-0000-4aaa-9000-1234567890ff",
  "id": "aaaaaaaa-0000-4aaa-9000-1234567890aa",
  "labels": [
    {
      "isoCode": "fra",
      "value": "Ancien chemin du hero",
    },
  ],
  "mainCommonToponymID": "bbbbbbbb-0000-4aaa-9000-1234567890bb",
  "meta": {
    "bal": {
      "cleInterop": "21286_0001_00001",
    },
  },
  "number": 1,
  "positions": [
    {
      "geometry": {
        "coordinates": [
          1,
          2,
        ],
        "type": "Point",
      },
      "type": "entrance",
    },
  ],
  "secondaryCommonToponymIDs": [
    "cccccccc-0000-4aaa-9000-1234567890cc",
    "cccccccc-1111-4aaa-9000-1234567890dd",
    "cccccccc-2222-4aaa-9000-1234567890ee",
  ],
  "updateDate": 2021-01-01T00:00:00.000Z,
}
`;

exports[`balAddrToBanAddr > should return BanAddress with multiple positions 1`] = `
{
  "certified": true,
  "districtID": "dddddddd-0000-4aaa-9000-1234567890ff",
  "id": "aaaaaaaa-0000-4aaa-9000-1234567890aa",
  "mainCommonToponymID": "bbbbbbbb-0000-4aaa-9000-1234567890bb",
  "meta": {
    "bal": {
      "cleInterop": "21286_0001_00001",
    },
  },
  "number": 1,
  "positions": [
    {
      "geometry": {
        "coordinates": [
          1,
          2,
        ],
        "type": "Point",
      },
      "type": "entrance",
    },
    {
      "geometry": {
        "coordinates": [
          3,
          4,
        ],
        "type": "Point",
      },
      "type": "other",
    },
  ],
  "secondaryCommonToponymIDs": [
    "cccccccc-0000-4aaa-9000-1234567890cc",
    "cccccccc-1111-4aaa-9000-1234567890dd",
    "cccccccc-2222-4aaa-9000-1234567890ee",
  ],
  "updateDate": 2021-01-01T00:00:00.000Z,
}
`;

exports[`balAddrToBanAddr > should return BanAddress with number '0' 1`] = `
{
  "certified": true,
  "districtID": "dddddddd-0000-4aaa-9000-1234567890ff",
  "id": "aaaaaaaa-0000-4aaa-9000-1234567890aa",
  "mainCommonToponymID": "bbbbbbbb-0000-4aaa-9000-1234567890bb",
  "meta": {
    "bal": {
      "cleInterop": "21286_0001_00001",
    },
  },
  "number": 0,
  "positions": [
    {
      "geometry": {
        "coordinates": [
          1,
          2,
        ],
        "type": "Point",
      },
      "type": "entrance",
    },
  ],
  "secondaryCommonToponymIDs": [
    "cccccccc-0000-4aaa-9000-1234567890cc",
    "cccccccc-1111-4aaa-9000-1234567890dd",
    "cccccccc-2222-4aaa-9000-1234567890ee",
  ],
  "updateDate": 2021-01-01T00:00:00.000Z,
}
`;

exports[`balAddrToBanAddr > should return BanAddress without BanID & BanTopoID 1`] = `
{
  "certified": true,
  "districtID": undefined,
  "id": undefined,
  "mainCommonToponymID": undefined,
  "meta": {
    "bal": {
      "cleInterop": "21286_0001_00001",
    },
  },
  "number": 1,
  "positions": [
    {
      "geometry": {
        "coordinates": [
          1,
          2,
        ],
        "type": "Point",
      },
      "type": "entrance",
    },
  ],
  "secondaryCommonToponymIDs": undefined,
  "updateDate": 2021-01-01T00:00:00.000Z,
}
`;

exports[`balAddrToBanAddr > should return BanToponym with multilingual label and personal config for default lang 1`] = `
{
  "certified": true,
  "districtID": undefined,
  "id": undefined,
  "labels": [
    {
      "isoCode": "eus",
      "value": "Antzinako Heroiaren Bidea",
    },
    {
      "isoCode": "fra",
      "value": "Ancien chemin du hero",
    },
  ],
  "mainCommonToponymID": "bbbbbbbb-0000-4aaa-9000-1234567890bb",
  "meta": {
    "bal": {
      "cleInterop": "21286_0001_00001",
    },
  },
  "number": 1,
  "positions": [
    {
      "geometry": {
        "coordinates": [
          1,
          2,
        ],
        "type": "Point",
      },
      "type": "entrance",
    },
  ],
  "secondaryCommonToponymIDs": undefined,
  "updateDate": 2021-01-01T00:00:00.000Z,
}
`;

exports[`balAddrToBanAddr > should return BanToponym with multilingual label, personal config for default lang  and overwrited default value 1`] = `
{
  "certified": true,
  "districtID": undefined,
  "id": undefined,
  "labels": [
    {
      "isoCode": "eus",
      "value": "Denboraren heroiaren antzinako bidea",
    },
    {
      "isoCode": "fra",
      "value": "Ancien chemin du hero du temps",
    },
  ],
  "mainCommonToponymID": "bbbbbbbb-0000-4aaa-9000-1234567890bb",
  "meta": {
    "bal": {
      "cleInterop": "21286_0001_00001",
    },
  },
  "number": 1,
  "positions": [
    {
      "geometry": {
        "coordinates": [
          1,
          2,
        ],
        "type": "Point",
      },
      "type": "entrance",
    },
  ],
  "secondaryCommonToponymIDs": undefined,
  "updateDate": 2021-01-01T00:00:00.000Z,
}
`;
